pyvista/core/filters/poly_data.py

```diff
@@ -1018,25 +1019,27 @@ def tube(
         scalars : str, optional
             Scalars array by which the radius varies.
 
-        capping : bool, optional
-            Turn on/off whether to cap the ends with polygons. Default
-            ``True``.
+        capping : bool, default: True
+            Turn on/off whether to cap the ends with polygons.
 
-        n_sides : int, optional
+        n_sides : int, default: 20
             Set the number of sides for the tube. Minimum of 3.
 
-        radius_factor : float, optional
+        radius_factor : float, default: 10
```

Comment:
And since I'm already asking for revisions: it's a bit dirty to claim this is a `float` and use `10` (if someone were to infer types from the default, they might think this is an int). For purity's sake we should probably change the default to `10.0`.

Reply:
Super, thanks for your feedback. This makes sense.

Out of curiosity, some parts of PyVista use type hints for the arguments (in addition to the types in the docstrings) and some not. Are type hints intentionally avoided, or is it just a matter of convenience?
I do not see this discussed in the Contributing docs.

Reply:
As far as I'm concerned typing happens to other people, and I personally find type hints a lot harder to read :sweat_smile: But at the same time I acknowledge that it can help downstream users using an IDE (I don't).

Apparently we have an older tracking issue for adding more type hints: https://github.com/pyvista/pyvista/issues/1136

And @tkoyama010 mentioned wanting to increase the coverage of type hints in https://github.com/pyvista/pyvista/discussions/2133#discussioncomment-2210496.

Reply:
Yeah, I understand, even if I consider myself to be more in the IDE/pro-type-hints camp, haha.
It is so great that both ways can happily co-exist in the codebase :) Thanks for the links to the issues.

Reply:
For completeness' sake here's Alex' take on the matter from an issue I overlooked on my first search https://github.com/pyvista/pyvista/pull/2904#issuecomment-1169968918:

> Let it be known that type annotations is still a work in progress and we don't have it in most places. adam-grant-hendry, if you feel up to it, you can feel free to add them. I'd do it slowly across several PRs. Tracking down `mypy` errors usually drives me up a wall.